tools/blue_deployment/helpers.js

```diff
@@ -0,0 +1,136 @@
+    }
+}
+
+/**
+ * Deploy the provided deployment objects, as provided by
+ * {@link ContractRegistry#toLiveTransactionObjects}, in sequence.
+ *
+ * @param {object} signer The signer used to sign the transactions.
+ *                 See {@link UnlockedWeb3Signer}.
+ * @param {object} sendWeb3 The web3 instance used to send the deployment transactions.
+ * @param {Array.<object>} deploymentObjects The deployment objects to deploy.
+ * @param {bool} options.log Whether to log the progress of the deployment.
+ * @param {string} options.gas The amount of gas to use for the deployment transactions.
+ *                 If omitted, it will be calculated via `web3.eth.estimateGas`.
+ * @param {string} options.gasPrice The gas price to use for the deployment transactions.
+ *                 If omitted, it will be calculated via `web3.eth.getGasPrice`.
+ */
+const deployContracts = async (
+    signer,
+    sendWeb3,
+    deploymentObjects,
+    options = {
+        log: false,
+    },
+) => {
+    // allows us to run promises in sequence
+    const asyncForEach = async (array, callback) => {
+        for (let index = 0; index < array.length; index++) {
+            await callback(array[index], index, array);
+        }
+    };
+
+    const _options = options;
+    const loggingActive = _options.log;
+    delete _options.log;
+
+    const contractAddresses = {};
+    await asyncForEach(deploymentObjects, async (deploymentObject) => {
+        let { gas, gasPrice } = _options;
+        if (!gas) {
+            gas = await sendWeb3.eth.estimateGas(deploymentObject.transactionObject);
+        }
+        if (!gasPrice) {
+            gasPrice = await sendWeb3.eth.getGasPrice();
+        }
+
+        return signer.signTransaction(Object.assign(
+            deploymentObject.transactionObject,
+            _options,
+            {
+                gas,
+                gasPrice,
+            },
+        ))
+            .then(signedTransaction => sendWeb3.eth.sendSignedTransaction(signedTransaction.raw))
+            .then((receipt) => {
+                if (loggingActive) {
+                    console.log(colors.dim(`Deploying ${deploymentObject.contractName} (tx ${receipt.transactionHash})`));
+                }
+                return waitForTransactionReceiptMined(sendWeb3, receipt.transactionHash, 10);
+            })
+            .then((receipt) => {
+                if (loggingActive) {
+                    console.log(colors.green(`Deployed ${colors.underline.green(deploymentObject.contractName)} at ${receipt.contractAddress}`));
+                    if (receipt.contractAddress.toLowerCase() !== deploymentObject.address) {
+                        console.warn(`"${deploymentObject.contractName}" was not deployed at the predicted address ${deploymentObject.address})`);
+                    }
+                }
+                contractAddresses[deploymentObject.contractName] = receipt.contractAddress.toLowerCase();
+            });
+    });
```

Comment:
Very long, can it be split?

Reply:
Yes, 104-126 Should probably be its own function